private/model/api/shape.go

```diff
@@ -130,6 +130,33 @@ func (s *Shape) GoTypeWithPkgName() string {
 	return goType(s, true)
 }
 
+// GenAccessors returns if the shape's reference should have setters and
+// getters generated.
+func (s *ShapeRef) UseIndirection() bool {
+	switch s.Shape.Type {
+	case "map", "list", "blob", "structure":
+		return false
+	}
+
+	if s.Streaming || s.Shape.Streaming {
```

Comment:
I thought of this, but didn't because its a bit harder to read. The `!` is really easy to miss.